crates/core/moonbase/src/lib.rs

```diff
@@ -128,77 +147,85 @@ impl Moonbase {
             return Ok(());
         }
 
-        let data: Response<ArtifactResponse> = parse_response(response.text().await?)?;
-        let error_message = match data {
-            Response::Failure { message, .. } => message,
-            _ => "Unknown failure!".into(),
-        };
-
         Err(MoonbaseError::ArtifactDownloadFailure(
             hash.to_string(),
-            error_message,
+            status
+                .canonical_reason()
+                .unwrap_or("Internal server error")
+                .to_owned(),
         ))
     }
 }
 
-// This is a stand-alone function so that we may run it in the background in a tokio thread,
+// This is a stand-alone function so that we may run it in a background Tokio thread,
 // and not have to worry about lifetime and borrow issues.
 pub async fn upload_artifact(
     auth_token: String,
     hash: String,
-    target_id: String,
     path: PathBuf,
     upload_url: Option<String>,
-) -> Result<Artifact, MoonbaseError> {
+) -> Result<(), MoonbaseError> {
     let file = fs::File::open(&path)
         .await
         .map_err(|e| map_io_to_fs_error(e, path.to_path_buf()))?;
-    let file_name = match path.file_name() {
-        Some(name) => name.to_string_lossy().to_string(),
-        None => format!("{}.tar.gz", hash),
-    };
-    let file_size = match file.metadata().await {
-        Ok(meta) => meta.len(),
-        Err(_) => 0,
-    };
     let file_stream = FramedRead::new(file, BytesCodec::new());
 
-    let form = Form::new().text("target", target_id.to_owned()).part(
-        "file",
-        Part::stream(Body::wrap_stream(file_stream))
-            .file_name(file_name.clone())
-            .mime_str("application/gzip")?,
-    );
-
     let request = if let Some(url) = upload_url {
-        reqwest::Client::new().post(url).multipart(form)
+        reqwest::Client::new()
+            .post(url)
+            .body(Body::wrap_stream(file_stream))
     } else {
         reqwest::Client::new()
-            .post(format!("{}/artifacts/{}/upload", get_host(), hash))
-            .multipart(form)
-            .bearer_auth(auth_token)
+            .post(endpoint(format!("artifacts/{}/upload", hash)))
```

Comment:
thought we were removing upload?

Reply:
im going to leave the logic in for now in case we need it in the future